src/aoc2020/day3.clj

```diff
@@ -0,0 +1,125 @@
+(ns aoc2020.day3
+  (:require [aoc2020.util :refer [file->seq]]))
+
+(defn coordination
+  [down-slope right-slope]
+  (let [row-s (take-nth down-slope (range))
+        col-s (take-nth right-slope (range))]
+    (map (fn [row-s col-s] [row-s col-s])
+      row-s col-s)))
+
+(defn tree-enctr?
+  [grid-s [row col]]
+  (let [col-size (count (get grid-s 1))
+        cur-col  (mod col col-size)
+        pos      (-> grid-s
+                   (get row)
+                   (get cur-col))]
+    (if (= pos \#)
+      true
+      false)))
+
+(defn total-tree-enctr-cnt
+  [grid-s & {:keys [right-slp down-slp]}]
```

Comment:
avoid varargs for passing options. It should only be (rarely) used for DSLs in libraries